src/mongo/integration/standalone/dbclient_test.cpp

```diff
@@ -1413,5 +1413,60 @@ TEST_F(DBClientTest, ConnectionStringWithTestDB) {
     }
 }
 
+std::size_t cursorCount(DBClientCursor* cursor) {
+    std::size_t count = 0;
+    while (cursor->more()) {
+        cursor->nextSafe();
+        ++count;
+    }
+    return count;
+}
+
+TEST_F(DBClientTest, LimitWithMultipleBatches) {
+    // We need to insert enough documents in to our collection such that we can require multiple
+    // batches to fill our limit.
+    std::size_t totalInserts = 25000;
+    std::size_t bulkSize = 1000;
+    for (std::size_t i = 0; i < totalInserts; i += bulkSize) {
+        BulkOperationBuilder bulk(&c, TEST_NS, false);
+        WriteResult result;
+        for (std::size_t j = 0; j < bulkSize; ++j) {
+            bulk.insert(BSON("i" << static_cast<long long>(i) << "j" << static_cast<long long>(j)
+                                 << "a"
+                                 << "bbbbbbbbbbbbbbbbbbbbbbbbbb"
+                                 << "c"
+                                 << "dddddddddddddddddddddddddddd"));
+        }
+        bulk.execute(&WriteConcern::acknowledged, &result);
+        ASSERT_EQUALS(static_cast<std::size_t>(result.nInserted()), bulkSize);
+    }
+
+    {
+        // A sanity check.
+        auto_ptr<DBClientCursor> limitQuery1 = c.query(TEST_NS, mongo::Query("{}"), 1);
```

Comment:
I see that this test changes the limit, but I think it is sticking with the default batch size? Is there a way we can explore extremal values of batchSize and ensure that things still work?